compiler/rustc_hir_pretty/src/lib.rs

```diff
@@ -1687,7 +1687,13 @@ impl<'a> State<'a> {
 
             let mut nonelided_generic_args: bool = false;
             let elide_lifetimes = generic_args.args.iter().all(|arg| match arg {
-                GenericArg::Lifetime(lt) => lt.is_elided(),
+                GenericArg::Lifetime(lt) => {
+                    let is_elided_lifetime = lt.is_elided();
+                    if !is_elided_lifetime {
+                        nonelided_generic_args = true;
+                    }
+                    is_elided_lifetime
```

Comment:
A bit messy here. I feel like if any better way.

Reply:
Could you replace this branch by changing the `if nonelided_generic_args` to `if nonelided_generic_args || !elide_lifetimes` below the loop?

Reply:
Yeah, that would work fine. But I have a concern that it would make `nonelided_generic_args` differ from what is inferred from its name.
Because in that case, `nonelided_generic_args` can be false, even when we have lifetime generic_args that are not elided.

Isn't it a concern that much? or do I understand wrong?

Reply:
`nonelided_generic_args` is only used in one place, just below, so I'm not really concerned. If you want to fix that, you can change the name.
Another possibility: make `lt.is_elided()` into an arm guard, and let the non-elided case fall through the `_ =>` case.
I'll let this to your judgement.

Reply:
> nonelided_generic_args is only used in one place, just below, so I'm not really concerned.

Ah, indeed.

> Another possibility: make lt.is_elided() into an arm guard, and let the non-elided case fall through the _ => case.

Thanks for the suggestion! I prefer this one a bit. But looks like non-elided case cannot fall to the `_ =>` case, so may be like this:

```rust
                GenericArg::Lifetime(lt) if lt.is_elided() => true,
                GenericArg::Lifetime(_) => {
                    nonelided_generic_args = true;
                    false
                }
                _ => {
                    nonelided_generic_args = true;
                    true
                }
```

Reply:
Made a change.